crates/control-interface/src/types.rs

```diff
@@ -3,7 +3,14 @@ use std::collections::HashMap;
 use anyhow::bail;
 use serde::{Deserialize, Serialize};
 
-/// One of a potential list of responses to an actor auction
+pub type ConstraintMap = std::collections::HashMap<String, String>;
+pub type AnnotationMap = std::collections::HashMap<String, String>;
+pub type KeyValueMap = std::collections::HashMap<String, String>;
+pub type LabelsMap = std::collections::HashMap<String, String>;
+pub type ProviderDescriptions = Vec<ProviderDescription>;
```

Comment:
Nit: My other comment around type aliases I could be convinced away from, but I _despise_ aliased Vecs like this 😆